src/store/api/worldUsers.ts

```diff
@@ -0,0 +1,231 @@
+  worldUserLocationsById: Record<string, WithId<UserLocation>>;
+}
+
+export const initialState: Readonly<WorldUsersData> = {
+  worldUsers: [],
+  worldUsersById: {},
+  worldUserLocationsById: {},
+};
+
+// TODO: https://redux-toolkit.js.org/rtk-query/api/createApi
+// TODO: https://redux-toolkit.js.org/rtk-query/usage/customizing-queries
+// TODO: https://redux-toolkit.js.org/rtk-query/usage/customizing-queries#streaming-data-with-no-initial-request
+// TODO: https://redux-toolkit.js.org/rtk-query/usage/customizing-queries#adding-meta-information-to-queries
+// TODO: https://redux-toolkit.js.org/rtk-query/usage/customizing-queries#constructing-a-dynamic-base-url-using-redux-state
+//   Can we use this to read the sovereignVenueId (or even currentVenueId or maybe all relatedVenueIds as needed) directly out of the state?
+//   And even if we can, do we want to? Or keep passing it in through WorldUsersApiArgs? (maybe the latter, as I think then it
+//     is used for/relates to the cache key for this data)
+// TODO: https://redux-toolkit.js.org/rtk-query/usage/migrating-to-rtk-query
+export const worldUsersApi = createApi({
+  reducerPath: "worldUsersApi",
+  baseQuery: fakeBaseQuery(),
+  endpoints: (builder) => ({
+    worldUsers: builder.query<WorldUsersData, WorldUsersApiArgs>({
+      // The query is not relevant here as the data will be provided via streaming updates.
+      // A queryFn returning an empty array is used, with contents being populated via
+      // streaming updates below as they are received.
+      queryFn: () => ({
+        data: initialState,
+        // TODO: we can also return error and/or meta here as well as data. Do we need to be able to?
+        // error: undefined,
+        // meta: undefined,
+      }),
+      async onCacheEntryAdded(
+        { relatedLocationIds },
+        { updateCachedData, cacheEntryRemoved }
+      ) {
+        // Used to hold the changes queued from the snapshot listener so that we can proces them in batches
+        const queuedChanges: firebase.firestore.DocumentChange<firebase.firestore.DocumentData>[] = [];
+
+        const processQueuedChanges = () => {
+          if (queuedChanges.length === 0) return;
+
+          updateCachedData((draft) => {
+            // We use splice here to remove all elements from the array and return them for processing
+            queuedChanges.splice(0).forEach(processUserDocChange(draft));
+          });
+        };
+
+        // TODO: move this interval into a proper const/config value somewhere
+        const processQueuedChangesIntervalId = setInterval(
+          processQueuedChanges,
+          5000
+        );
+
+        const worldUsersQuery = firebase
+          .firestore()
+          .collection("users")
+          .where("enteredVenueIds", "array-contains-any", relatedLocationIds);
```

Comment:
How often is this going to be called? Is `array-contains-any` going to work properly if `relatedLocationIds` has more than 10 entries?

Reply:
What do you mean by ‘how many times will it be called’?

Nope, it would max out at 10, but I think the naming is sort of legacy as I think we generally work on the assumption that all users will have entered the sovereignVenue.

I’m honestly not sure if we even need to have the current venue id in there either; but I didn’t want to change the existing logic too much here.

With the proper ‘world architecture’ refactor I expect this should be a lot more obvious/straightforward.